### PR TITLE
making tests parallel

### LINKED ISSUE #1: Make e2e tests run in parallel
End to end tests run sequentially right now, we should make them run in parallel.

### PR DESCRIPTION
**Pull request by dgkanatsios**
Closes #1 

This PR completely upgrades the end to end test infrastructure by utilizing ginkgo/gomega libraries and parallelism.
Also:
- upgrading e2e kind cluster to Kubernetes 1.22.5
- increasing the resource limits/requests of the controller since they were too low
- fixing an issue on the GameServerBuild controller where it would ignore downscaling if the GameServerBuild has GameServers stuck in initializing (corner case but still)
- modifies a GameServerBuild controller unit test to use .Patch instead of .Update (it would rarely create conflicts)

### CODE REVIEW

File: e2e/run.sh [hunk 1]
```diff
@@ -44,11 +44,8 @@ kubectl wait --for=condition=ready --timeout=300s pod -n thundernetes-system -l
...
 echo "-----Running end to end tests-----"
 cd cmd/e2e
-go build && ./e2e ${IMAGE_NAME_NETCORE_SAMPLE}:${IMAGE_TAG} && rm e2e
-
-echo "-----Running GameServer API tests-----"
-# create the gameserverapi namespace for the GameServer API tests
+# create the test namespaces
 kubectl create namespace gameserverapi
-go get github.com/onsi/ginkgo/v2
+kubectl create namespace mynamespace
```
Comment (khaines) [inline e2e/run.sh:49]: Nit: consider a better name ... this feels like a copy/paste of an example

Comment (dgkanatsios) [reply e2e/run.sh]: changed, thanks!

**Review by khaines — APPROVED** [review]
Wow, thanks for this change @dgkanatsios ! this really helps running the tests quickly
